extensions/notebook/src/book/bookTreeView.ts

```diff
@@ -74,6 +75,39 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
 			}
 		} catch (e) {
 			vscode.window.showErrorMessage(loc.openFileError(bookPath, e instanceof Error ? e.message : e));
+		} finally {
+			// add file watcher on toc file.
+			fsw.watchFile(path.join(bookPath, '_data', 'toc.yml'), async (curr, prev) => {
+				if (curr.mtime > prev.mtime) {
+					let index = this.books.findIndex(book => book.bookPath === bookPath);
+					await this.books[index].initializeContents().then(() => {
+						this._onDidChangeTreeData.fire(this.books[index].bookItems[0]);
+					});
+					this._onDidChangeTreeData.fire();
+				}
+			});
+		}
+	}
+
+	async closeBook(book?: BookTreeItem): Promise<void> {
+		// remove book from the saved books
+		let deletedBook: BookModel;
+		try {
+			if (book) {
+				let index: number = this.books.indexOf(this.books.find(b => b.bookPath.replace(/\\/g, '/') === book.root));
```

Comment:
It's too bad we have to do this kind of check - shouldn't the tree item know about the model anyways so that you could just do a direct lookup?